auth-api/src/auth_api/services/affiliation.py

```diff
@@ -0,0 +1,166 @@
+
+    @property
+    def org(self):
+        """Return the org for this Affiliation."""
+        return self._model.org
+
+    @org.setter
+    def org(self, value: int):
+        """Set the org for this Affiliation."""
+        self._model.org = value
+
+    @ServiceTracing.disable_tracing
+    def as_dict(self):
+        """Return the affiliation as a python dictionary.
+
+        None fields are not included in the dictionary.
+        """
+        affiliation_schema = AffiliationSchema()
+        obj = affiliation_schema.dump(self._model, many=False)
+        return obj
+
+    @staticmethod
+    def find_affiliation_by_ids(org_id, affiliation_id):
+        """Given an org_id and affiliation_id, this will return the corresponding affiliation or None."""
+        if not org_id or not affiliation_id:
+            return None
+
+        affiliation_model = AffiliationModel.find_affiliation_by_ids(org_id, affiliation_id)
+
+        if not affiliation_model:
+            raise BusinessException(Error.DATA_NOT_FOUND, None)
+
+        return Affiliation(affiliation_model).as_dict()
+
+    @staticmethod
+    def find_affiliations_by_org_id(org_id):
+        """Given an org_id, this will return the corresponding affiliations or None."""
+        current_app.logger.debug('<find_affiliations_by_org_id for org_id {}'.format(org_id))
+        if not org_id:
+            return None
+
+        org = OrgService.find_by_org_id(org_id)
+        if org is None:
+            raise BusinessException(Error.DATA_NOT_FOUND, None)
+
+        data = {'items': []}
+        affiliation_models = AffiliationModel.find_affiliations_by_org_id(int(org_id))
+        if affiliation_models is None:
+            raise BusinessException(Error.DATA_NOT_FOUND, None)
+
+        for affiliation_model in affiliation_models:
+            if affiliation_model:
+                data['items'].append(Affiliation(affiliation_model).as_dict())
+        current_app.logger.debug('>find_affiliations_by_org_id')
+
+        return data
+
+    @staticmethod
+    def create_affiliation(org_id, entity_info: Dict[str, Any]):
+        """Create an Affiliation."""
+
+        # Validate if org_id is valid by calling Org Service.
+        org = OrgService.find_by_org_id(org_id)
+        if org is None:
+            raise BusinessException(Error.DATA_NOT_FOUND, None)
+
+        entity = EntityService.find_by_entity_id(entity_info.get('entity'))
+        if entity is None:
+            raise BusinessException(Error.DATA_NOT_FOUND, None)
+
+        affiliation = AffiliationModel()
+        affiliation.entity = entity.as_dict()['id']
+        affiliation.org = org_id
+
+        affiliation.save()
+        affiliation = Affiliation(affiliation)
+
+        return affiliation.as_dict()
+
+    @staticmethod
+    def update_affiliation(org_id, affiliation_id, entity_info: Dict[str, Any]):
+        """Update an Affiliation."""
+        affiliation = AffiliationModel.find_by_affiliation_id(int(affiliation_id))
```

Comment:
This cast to `int` should not be necessary.  It could also throw a runtime exception if the passed `affiliation_id` is not castable to an `int`, which would need to be caught.